homework27/src/ba/bitcamp/homework/tasks/Task2.java

```
@@ -0,0 +1,164 @@
+	private static ArrayList<Worker> workers;
+	private static int a = 0;
+	private static int b = 10;
+
+	public static void main(String[] args) {
+		new Task2();
+	}
+
+	/**
+	 * Default constructor that initializes and sets gui components
+	 */
+	public Task2() {
+		importPicture();
+		panel = new MyPanel();
+		add(panel);
+
+		setSize(1920, 1080);
+		setResizable(false);
+		setLocationRelativeTo(null);
+		setTitle("Negativ Task");
+		setDefaultCloseOperation(EXIT_ON_CLOSE);
+		setVisible(true);
+
+		queue = new LinkedBlockingQueue<Runnable>();
+		// Adding new tasks to queue
+		for (int i = 0; i < 150; i++) {
+
+			queue.add(new Task(a, b));
+			a += 10;
+			b += 10;
+
+		}
+
+		workers = new ArrayList<>();
+		// Adding workers
+		for (int i = 0; i < 8; i++) {
+			Worker s = new Worker();
+			s.start();
+			try {
+				s.join();
+			} catch (InterruptedException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+			workers.add(s);
+		}
+
+	}
+
+	/**
+	 * This class represents task that is proceeded to one of eight workers
+	 * 
+	 *
+	 */
+	static class Task implements Runnable {
+
+		private int a;
+		private int b;
+
+		public Task(int a, int b) {
+			this.a = a;
+			this.b = b;
+
+		}
+
+		@Override
+		public void run() {
+
+			for (int i = a; i < b; i++) {
+				for (int j = 0; j < img.getHeight(); j++) {
+					int pixel = img.getRGB(i, j);
+					img.setRGB(i, j, 255 - pixel);
+
+				}
+
+			}
+
+		}
+
+	}
+
+	/**
+	 * Class that is used for taking jobs until queue is empty
+	 * 
+
+	 *
+	 */
+	static class Worker extends Thread {
+		@Override
+		public void run() {
+			while (!queue.isEmpty()) {
+				try {
+					Runnable job = queue.take();
+
+					job.run();
+				} catch (InterruptedException e) {
+					break;
+				}
+			}
+		}
+	}
+
+	/**
+	 * Method that reads picture from file
+	 */
+	public void importPicture() {
+		try {
+			img = ImageIO.read(new File("grb1.jpg"));
+
+		} catch (IOException e) {
+
+			e.printStackTrace();
```

Again, exception is ignored.